core/src/main/scala/kafka/tools/ConsoleProducer.scala

```diff
@@ -29,8 +25,16 @@ import kafka.utils.{CommandDefaultOptions, CommandLineUtils, Exit, ToolsUtils}
 import org.apache.kafka.clients.producer.internals.ErrorLoggingCallback
 import org.apache.kafka.clients.producer.{KafkaProducer, ProducerConfig, ProducerRecord}
 import org.apache.kafka.common.KafkaException
+import org.apache.kafka.common.header.Header
+import org.apache.kafka.common.header.internals.RecordHeader
```

Comment:
Ideally we should avoid using this internal class. Can we instead use https://kafka.apache.org/30/javadoc/org/apache/kafka/common/header/Headers.html?

Reply:
Hmmm, thanks, isn't it the case that RecordHeader is the only implementation of Headers?

Or in other words what implementation should I return on line 353?
https://github.com/apache/kafka/pull/11456/files#diff-3b7393aaaffd466b3bfc53c8d95109a206442160b878bbdaa0b0332257afaebaR353

Reply:
The alternative is to create a `ProducerRecord` and then do `record.headers.add(String, byte[])`